openhcl/virt_mshv_vtl/src/processor/tdx/mod.rs

```diff
@@ -1745,16 +1772,15 @@ impl UhProcessor<'_, TdxBacked> {
         intercepted_vtl: GuestVtl,
     ) -> Result<(), MsrError> {
         match msr {
-            msr @ hvdef::HV_X64_MSR_GUEST_OS_ID => self.hv[intercepted_vtl]
-                .as_mut()
-                .unwrap()
-                .msr_write(msr, value)?,
+            msr @ hvdef::HV_X64_MSR_GUEST_OS_ID => {
+                self.backing.cvm.hv[intercepted_vtl].msr_write(msr, value)?
+            }
             _ => {
-                self.untrusted_synic.as_mut().unwrap().write_nontimer_msr(
-                    &self.partition.gm[GuestVtl::Vtl0],
-                    msr,
-                    value,
-                )?;
+                self.backing
+                    .untrusted_synic
+                    .as_mut()
+                    .unwrap()
```

Comment:
Maybe add an expect here about why we know this is non-None.

Reply:
How's this?